x-pack/metricbeat/module/mssql/connection.go

```diff
@@ -21,9 +21,9 @@ func NewConnection(uri string) (*sql.DB, error) {
 
 	// Check the connection before executing all queries to reduce the number
 	// of connection errors that we might encounter.
-	if err = db.Ping(); err != nil {
-		err = fmt.Errorf("error doing ping to db: %w", err)
-	}
+	//if err = db.Ping(); err != nil {
+	//	err = fmt.Errorf("error doing ping to db: %w", err)
+	//}
```

Comment:
[Blocker]
I'm concerned it might not a safe option to remove this ping. `sql.Open` does not guarantee the connection to the DB works:

>// Open may just validate its arguments without creating a connection
// to the database. To verify that the data source name is valid, call
// [DB.Ping].

and checking the driver we use, I only saw them parsing the connection string, not actually checking the connection.

In addition to that, the calls using `mssql.NewConnection` panic if there is an error. Thus, if I'm not missing something, the current design requires this check to ensure the connection works. If there is an error in the connection string, not a transient error that will fix itself, it'll either never be auto-fixed, the metrics will keeping throwing errors that might even be harder to understand or debug than if they were coming from `mssql.NewConnection`.

https://github.com/elastic/beats/blob/b872d5e21ebb07e30ec7a6e9a593c682e50e927a/x-pack/metricbeat/module/mssql/performance/performance.go#L29-L30

https://github.com/elastic/beats/blob/2fec2d91f06774cddb9b7b974d4a10144b0fd786/x-pack/metricbeat/module/mssql/transaction_log/transaction_log.go#L40-L41

I get it might better not make this check here, but then the metricsets using this connection should also be modified to better handle that situation.

Reply:
This was only added for testing purposes, to see if it fixes the issue. Last thing I want to do is change code to update the golang version. It didn't fix the issue, so it should be reverted anyway.